api/swagger.yaml

```diff
@@ -0,0 +1,282 @@
+openapi: "3.0.2"
+info:
+  title: Chat Dancer
+  version: "1.0"
+servers:
+  - url: 'https://chat-dancier.net'
+  - url: 'http://localhost:8080'
+paths:
+  /chats:
+    get:
+      parameters:
+        - $ref: '#/components/parameters/user_id'
+      summary: returns a list of all chats of a user. The list endpoint does only return the latest message, the messages array will only be filled on fetching messages for a chat
```

Comment:
I would say: this endpoint, instead of the list endpoint. 
The latter one, made me feel you are referencing another endpoint (the list one).

Then I would also say:
You can get all messages of a chat wie another endpoint.
Sound a little bit confusing, saying the message array will only be filled...
You are not filling the message array with the list endpoint, right?

Reply:
My intention was that the list endpoint is only used to get the data for the overview of chats. If the user then clicks on a chat, the messages for this chat are loaded. 
Yes, I will rephrase it a bit ;)